### DISCUSSION TITLE
What is the "Free Supercharging" toggle supposed to do?

### DISCUSSION THREAD

**Discussion by JSaterdalen** [teslamate-org/teslamate#1530] [Q&A] [answered]
Supercharging costs always show up as $0 unless I go in and manually add a cost, with Free Supercharging disabled. What is the benefit of the Free Supercharging toggle if it doesn't add charging costs either way? Is it there for some other tracking purpose?

**Comment by adriankumpf** [comment, marked as answer]
Yes, that is the purpose. Supercharges are set to $0 instead of the cost being empty when Free Supercharging is enabled.

**Reply by JSaterdalen** [reply]
That's why I'm asking—even with Free Supercharging disabled I don't have a cost showing up. Sounds like that's not intentional and I should look into as a bug?

They currently show up as null instead of 0.
![Screen Shot 2021-04-16 at 9 36 04 AM](https://user-images.githubusercontent.com/6761188/115040586-374d5f80-9e97-11eb-8a1a-a6f1d526ded3.png)

**Reply by ngardiner** [reply]
You would need to set the cost. TeslaMate doesn't know what individual superchargers cost.

**Reply by JSaterdalen** [reply]
Thanks for the clarification. Still trying to grasp everything that Teslamate can do!
